src/main/java/ua/tc/marketplace/model/dto/ad/CreateAdDto.java

```diff
@@ -16,23 +16,14 @@
  * <p>Validation constraints are applied to ensure data integrity and consistency.
  */
 public record CreateAdDto(
-    @NotNull(message = "Author ID cannot be null")
-    Long authorId,
-
+    @NotNull(message = "Author ID cannot be null") Long authorId,
     @NotEmpty(message = "Title cannot be empty")
-    @Size(max = 100, message = "Title cannot be longer than 100 characters")
-    String title,
-
-    @NotEmpty(message = "Description cannot be empty")
-    String description,
-
+        @Size(max = 100, message = "Title cannot be longer than 100 characters")
+        String title,
+    @NotEmpty(message = "Description cannot be empty") String description,
     @NotNull(message = "Price cannot be null")
-    @DecimalMin(value = "0.0", message = "Price must be greater than zero")
-    BigDecimal price,
-
-    @NotNull(message = "Photo files cannot be null")
-    MultipartFile[] photoFiles,
-
-    @NotNull(message = "Category ID cannot be null")
-    Long categoryId
-) {}
+        @DecimalMin(value = "0.0", message = "Price must be zero or positive")
+        BigDecimal price,
+    @NotNull(message = "Photo files cannot be null") MultipartFile[] photoFiles,
+    @NotEmpty(message = "Attributes cannot be empty") String adAttributes,
```

Comment:
Why do you receive adAttributes as a String and not as List<AdAttributeDto> for example?

Reply:
it is because we need to handle test fields, multipart files array and json object in one request. And to use `form-data` for this purpose. And when we do so, Spring automatically do not parse all of these data types in one object. So i found a solution when we place json object into string field and then parse it. That why it is string, inside which a json object is located. This json object contains an array of adAttributes